packages/wonder-blocks-data/src/util/in-memory-cache.js

```diff
@@ -0,0 +1,142 @@
+// @flow
+import {KindError, Errors, clone} from "@khanacademy/wonder-stuff-core";
+import type {ValidData, ScopedCache} from "./types.js";
+
+/**
+ * Describe an in-memory cache.
+ */
+export class InMemoryCache {
+    _cache: ScopedCache;
+
+    constructor(initialCache: ScopedCache = Object.freeze({})) {
+        try {
+            this._cache = clone(initialCache);
+        } catch (e) {
+            throw new KindError(
+                `An error occurred trying to initialize from a response cache snapshot: ${e}`,
+                Errors.InvalidInput,
+            );
+        }
+    }
+
+    /**
+     * Indicate if this cache is being used or not.
+     *
+     * When the cache has entries, returns `true`; otherwise, returns `false`.
+     */
+    get inUse(): boolean {
+        return Object.keys(this._cache).length > 0;
+    }
+
+    /**
+     * Set a value in the cache.
+     */
+    set: <TValue: ValidData>(scope: string, id: string, value: TValue) => void =
+        <TValue: ValidData>(scope, id, value: TValue): void => {
+            if (!id || typeof id !== "string") {
+                throw new KindError(
+                    "id must be non-empty string",
+                    Errors.InvalidInput,
+                );
+            }
+
+            if (!scope || typeof scope !== "string") {
+                throw new KindError(
+                    "scope must be non-empty string",
+                    Errors.InvalidInput,
+                );
+            }
+
+            if (typeof value === "function") {
+                throw new KindError(
+                    "value must be a non-function value",
+                    Errors.InvalidInput,
+                );
+            }
+
+            this._cache[scope] = this._cache[scope] ?? {};
+            this._cache[scope][id] = Object.freeze(clone(value));
+        };
+
+    /**
+     * Retrieve a value from the cache.
+     */
+    get: (scope: string, id: string) => ?ValidData = (
+        scope,
+        id,
+    ): ?ValidData => {
+        return this._cache[scope]?.[id] ?? null;
+    };
+
+    /**
+     * Purge an item from the cache.
+     */
+    purge: (scope: string, id: string) => void = (scope, id) => {
```

Comment:
In the other PR I seem to remember the equivalent of this function would return `false` if there was nothing to purge and `true` if purging was successful.  I'm curious why you dropped that behavior?

Reply:
Because nothing was using it and it was otherwise just bloating the code.